src/components/ComponentListView/ComponentListViewPage.tsx

```diff
@@ -51,48 +53,95 @@ export const ComponentListViewPage: React.FC<ComponentListViewPageProps> = ({ ap
 
   const onClearFilters = () => setNameFilter('');
   const onNameInput = (name: string) => setNameFilter(name);
+  const onSelectedColumnChange = (
+    checked: boolean,
+    event: React.FormEvent<HTMLInputElement>,
+  ): void => {
+    const updatedCheckedColumns = new Set<string>(selectedComponents);
+    const selectedItem = event?.currentTarget?.name;
+    updatedCheckedColumns.has(selectedItem)
+      ? updatedCheckedColumns.delete(selectedItem)
+      : updatedCheckedColumns.add(selectedItem);
+    setSelectedComponents(updatedCheckedColumns);
+  };
+
+  const deleteSelectedRows = async () => {
+    await Promise.all(
+      Array.from(selectedComponents).map((component) =>
+        deleteComponent(component, application, namespace),
+      ),
+    )
```

Comment:
These callbacks are making use of component state, would be good to memoize them.